src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -163,6 +163,10 @@ fn check_rvalue(
                 _ => check_operand(tcx, mir, operand, span),
             }
         }
+        Rvalue::Cast(_, _, _) => Err((
+            span,
+            "only int casts are allowed in const fn".into(),
```

Comment:
"int casts" doesn't seem like the right description here, not all `Misc` casts involved integers -- do they?

There are not so many kinds of casts, so depending on that you could make it say "unsizing casts" or "function pointer casts".

Reply:
Also, can you add a test case for this code branch?

Reply:
I'm not sure if I can, but I'll try

Reply:
Don't let it be a blocker, but it'd be nice to have error cases so that later we can give them all their own error code and a good extended description.

Reply:
I managed to get to the unsizing errors. Everything else errors out early due to fn pointers not being allowed